### PR TITLE
Fix : event log page is very slow for non-admins

### PR DESCRIPTION
**Pull request by TamazC**
## Description

This PR removes first and last page links in event logs page in order to increase page performance

**Fixes** # (issue)
MON-14657

## Type of change

- [ ] Patch fixing an issue (non-breaking change)
- [x] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software

## Target serie

- [ ] 21.10.x
- [ ] 22.04.x
- [ ] 22.10.x
- [x] 23.04.x (master)

<h2> How this pull request can be tested ? </h2>

Create a huge data in event logs and render on event logs page.

#### Community contributors & Centreon team

- [x] I have followed the **coding style guidelines** provided by Centreon
- [x] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [x] I have commented my code, especially **hard-to-understand areas** of the PR.
- [x] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

File: centreon/www/include/eventLogs/xml/PaginationRenderer.php [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+ *
+ * For more information : contact@centreon.com
+ *
+ */
+
+declare(strict_types=1);
+
+include_once _CENTREON_PATH_ . 'www/class/centreonXML.class.php';
+
+class PaginationRenderer
```
Comment (callapa) [inline centreon/www/include/eventLogs/xml/PaginationRenderer.php:25]:
```suggestion
class XmlPaginationRenderer
```

**Review by callapa — COMMENTED** [review]
Please add the PHP Doc

File: centreon/www/include/eventLogs/xml/data.php [hunk 2]
```diff
@@ -740,100 +744,20 @@
...
-    }
+    $paginationRenderer = new PaginationRenderer($buffer);
+    $paginationRenderer->render($pages);
 
     /*
      * Full Request
      */
     $cpts = 0;
-    foreach ($logs as $log) {
+    foreach (array_slice($logs, 0, $limit) as $log) {
```
Comment (abuathier) [inline centreon/www/include/eventLogs/xml/data.php:760]: Please put a comment to explain `array_slice`.

File: centreon/www/include/eventLogs/xml/data.php [hunk 3]
```diff
@@ -233,8 +236,8 @@
 }
 
 // binding limit value
-$limit = $inputs['limit'];
-$num = $inputs['num'];
+$num = filter_var($inputs['num'], FILTER_VALIDATE_INT, ['options' => ['default' => 0, 'min_range' => 0]]);
+$limit = filter_var($inputs['limit'], FILTER_VALIDATE_INT, ['options' => ['default' => 30]]);
```
Comment (abuathier) [inline centreon/www/include/eventLogs/xml/data.php:240]: Maybe rename as `pageSize` and not `limit` to be less confused.

**Review by abuathier — APPROVED** [review]

**Review by callapa — APPROVED** [review]

**Pull request merged**